packages/core/data-transfer/src/strapi/providers/local-destination/strategies/restore/links.ts

```diff
@@ -16,14 +35,26 @@ export const createLinksWriteStream = (
         const { left, right } = link;
         const query = createLinkQuery(strapi, trx);
 
+        const originalLeftRef = left.ref;
+        const originalRightRef = right.ref;
+
         // Map IDs if needed
-        left.ref = mapID(left.type, left.ref) ?? left.ref;
-        right.ref = mapID(right.type, right.ref) ?? right.ref;
+        left.ref = mapID(left.type, originalLeftRef) ?? originalLeftRef;
+        right.ref = mapID(right.type, originalRightRef) ?? originalRightRef;
 
         try {
           await query().insert(link);
         } catch (e) {
           if (e instanceof Error) {
+            if (
+              // e.message.toLowerCase().includes('foreign key constraint') ||
+              isForeignKeyConstraintError(e)
```

Comment:
```suggestion
              isForeignKeyConstraintError(e)
```

Reply:
Actually, thanks for that, but it's necessary in order to make the condition stronger, just in case some database error codes not match the ones I defined in `isForeignKeyConstraintError`